pkg/didcomm/common/service/destination.go

```diff
@@ -54,10 +64,67 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 
 	// TODO ensure recipient keys are did:key's
 	//  https://github.com/hyperledger/aries-framework-go/issues/1604
+	// above TODO is TBD - for now added KMS encryption key to support new Authcrypt/Anoncrypt
+
+	var recipientEncKeys []*composite.PublicKey
+
+	for _, recKey := range didCommService.RecipientKeys {
+		id, err := computeVerKID([]byte(recKey))
+		if err != nil {
+			return nil, err
+		}
+
+		encKey, encKeyFound := diddoc.LookupEncryptionKey(id, didDoc)
+		if encKeyFound {
+			enc, er := extractECKey(encKey.JSONWebKey())
+			if er != nil {
+				return nil, er
+			}
+
+			recipientEncKeys = append(recipientEncKeys, enc)
+		}
+
+	}
 
 	return &Destination{
-		RecipientKeys:   didCommService.RecipientKeys,
-		ServiceEndpoint: didCommService.ServiceEndpoint,
-		RoutingKeys:     didCommService.RoutingKeys,
+		RecipientKeys:    didCommService.RecipientKeys,
+		ServiceEndpoint:  didCommService.ServiceEndpoint,
+		RoutingKeys:      didCommService.RoutingKeys,
+		RecipientEncKeys: recipientEncKeys,
 	}, nil
 }
+
+func extractECKey(encKeyJWK *ariesjose.JWK) (*composite.PublicKey, error) {
+	tp, err := encKeyJWK.Thumbprint(sha256.Size)
+	if err != nil {
+		return nil, err
+	}
+
+	kid := base64.URLEncoding.EncodeToString(tp)
+
+	switch pubKey := encKeyJWK.Key.(type) {
+	case *ecdsa.PublicKey:
+		return &composite.PublicKey{
+			KID:   kid,
+			X:     pubKey.X.Bytes(),
+			Y:     pubKey.Y.Bytes(),
+			Curve: pubKey.Curve.Params().Name,
+		}, nil
+	default:
+		return nil, errors.New("")
```

Comment:
error message

Reply:
must have been very late and forgot to add a message..

Reply:
updated